events/serializers.py

```diff
@@ -953,6 +953,39 @@ def validate_keywords(self, keywords):
     def validate_audience(self, audiences):
         return self.validate_keywords(audiences)
 
+    def validate_external_links(self, value):
+        if not value:
+            return value
+
+        for index, link in enumerate(value):
+            # clean link text fields
+            value[index] = clean_text_fields(link)
+
+        checked_values = set()
+        for data in value:
+            name = data.get("name", "")
+            language = data["language"].pk
+            link = data["link"]
+
+            unique_link = f"{name}-{language}-{link}".lower()
+            if unique_link in checked_values:
+                raise serializers.ValidationError(
+                    {
+                        "name": _("External link with name %(name)s already exists.")
```

Comment:
From what I understand the links are always overwritten so if there would be some existing links they would be wiped if they weren't in the data. That in mind the text could be slightly modified - the `already exists` part specially, since it may or may not be existing but rather is given through the form. 
So something like `Duplicate link given` or something like that.

Reply:
This is a good suggestion. The form "... already exists" is used in some other similar duplicate checks, but I agree that it can be a bit confusing or misleading. I changed it as suggested.